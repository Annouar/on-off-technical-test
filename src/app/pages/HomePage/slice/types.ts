import { PhoneLog } from 'types/PhoneLogs';

export interface ListingState {
  loading: boolean;
  phoneLogs?: PhoneLog[];
}
