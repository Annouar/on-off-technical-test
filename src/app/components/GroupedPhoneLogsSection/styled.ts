import styled from 'styled-components/macro';

export const LogsContainer = styled.div`
  &>*: not(: first-child) {
    margin-top: 20px;
  }
`;
