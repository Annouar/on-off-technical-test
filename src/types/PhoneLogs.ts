export interface PhoneLog {
  id: string;
  phoneType: 'Mobile' | 'Work' | 'Home';
  phoneNumber: string;
  date: string;
}

export type GroupedLogs = Record<string, PhoneLog[]>;
