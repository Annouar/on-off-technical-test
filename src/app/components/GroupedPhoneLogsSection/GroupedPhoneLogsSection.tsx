import React, { FC } from 'react';
import { PhoneLog } from 'types/PhoneLogs';

import { PhoneLogCard } from 'app/components/PhoneLogCard/PhoneLogCard';

import { LogsContainer } from './styled';

interface Props {
  title: string;
  phoneLogs: PhoneLog[];
}

export const GroupedPhoneLogsSection: FC<Props> = ({ title, phoneLogs }) => {
  // TODO: profile this to know if useCallback is usefull here
  const sortedLogs = phoneLogs.sort(
    (a, b) => new Date(b.date).getTime() - new Date(a.date).getTime(),
  );

  return (
    <>
      <p>{title}</p>
      <LogsContainer>
        {sortedLogs.map(log => (
          <PhoneLogCard
            key={log.id}
            phoneNumber={log.phoneNumber}
            phoneType={log.phoneType}
            date={log.date}
          />
        ))}
      </LogsContainer>
    </>
  );
};
