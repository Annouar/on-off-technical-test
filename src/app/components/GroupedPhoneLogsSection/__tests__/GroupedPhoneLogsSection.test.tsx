import * as React from 'react';
import { render } from '@testing-library/react';

import { GroupedPhoneLogsSection } from '../GroupedPhoneLogsSection';
import { PhoneLog } from 'types/PhoneLogs';

describe('<GroupedPhoneLogsSection  />', () => {
  it('should match snapshot', () => {
    const phoneLogs = [
      {
        phoneNumber: '337899008978',
        date: '2021-04-06T18:20:06+0000',
        id: '7',
        phoneType: 'Mobile',
      },
      {
        phoneNumber: '33566787899',
        date: '2021-05-06T10:20:06+0000',
        id: '8',
        phoneType: 'Mobile',
      },
      {
        phoneNumber: '33566787899',
        date: '2021-04-06T10:20:06+0000',
        id: '9',
        phoneType: 'Mobile',
      },
    ] as PhoneLog[];

    const loadingIndicator = render(
      <GroupedPhoneLogsSection title="foo" phoneLogs={phoneLogs} />,
    );

    expect(loadingIndicator.container.firstChild).toMatchSnapshot();
  });
});
