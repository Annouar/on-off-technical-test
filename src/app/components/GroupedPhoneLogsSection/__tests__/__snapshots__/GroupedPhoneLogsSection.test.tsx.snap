// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroupedPhoneLogsSection  /> should match snapshot 1`] = `
<p>
  foo
</p>
`;
